core/lib/dal/src/transactions_web3_dal.rs

```diff
@@ -78,21 +78,17 @@ impl TransactionsWeb3Dal<'_, '_> {
             .fetch_optional(self.storage.conn())
             .await?
             .map(|db_row| {
-                let status = match (db_row.block_number, db_row.error) {
-                    (_, Some(_)) => Some(U64::from(0)),
-                    (Some(_), None) => Some(U64::from(1)),
-                    // tx not executed yet
-                    _ => None,
-                };
+                let status = db_row.error.map(|_| U64::zero()).unwrap_or_else(U64::one);
```

Comment:
unwrap_or_else(U64::one) looks weirds, it should be smth like `expect("Mined transaction must have status")`

Reply:
no, we unwrap the error.

Reply:
it's just `if none then zero else one  `